lib/src/main/java/io/ably/lib/realtime/ChannelBase.java

```diff
@@ -186,6 +193,40 @@ void attach(boolean forceReattach, CompletionListener listener) {
         attachWithTimeout(forceReattach, listener);
     }
 
+    /**
+     * This method carries queued messages accumulated on connection manager while the channel
+     * isn't attached yet. It's added in the queue here and start a new attach call
+     * */
+    void reattach(List<QueuedMessage> messagesToTransfer) {
+        state = ChannelState.attaching;
+        if (messagesToTransfer != null) {
+            for (QueuedMessage queuedMessage : messagesToTransfer) {
+                if (queuedMessage.msg.action == Action.message) {
+
+                    queuedMessages.add(queuedMessage);
+                }else if (queuedMessage.msg.action == Action.presence) {
+                    PresenceMessage[] presenceMessages = queuedMessage.msg.presence;
+                    if (presenceMessages != null && presenceMessages.length > 0){
+                        for (PresenceMessage presenceMessage : presenceMessages) {
+                            String clientId;
+                            try {
+                                clientId = ably.auth.checkClientId(presenceMessage, false, true);
```

Comment:
why is this being checked here? If the message had previously been queued, don't we know that the `clientId` is valid?

Reply:
I took this from an existing code 
https://github.com/ably/ably-java/blob/1c41924f08dffb1e94adc4ee730584b14f16103e/lib/src/main/java/io/ably/lib/realtime/Presence.java#L743
As I did not fully know the condition of clientId being valid here, I thought this check might be sensible. But I can understand why it's not needed here.
Change: c41824dc911210265ea8f3b55c998206f74ce953